fearless/Common/Services/PayoutRewardsService/PayoutValidatorsForNominatorFactory.swift

```diff
@@ -5,225 +5,156 @@ import IrohaCrypto
+        let requestFactory = createRequestFactory(historyRange: historyRange)
+        let resultFactory = createResultFactory()
 
-            let resultSet = allSets.reduce(into: Set<AccountId>()) { result, item in
-                result = result.union(item)
-            }
+        let operation = NetworkOperation(requestFactory: requestFactory, resultFactory: resultFactory)
+        return CompoundOperationWrapper(targetOperation: operation)
+    }
 
-            return Array(resultSet)
+    private func createRequestFactory(
+        historyRange: @escaping () -> ChainHistoryRange?
+    ) -> NetworkRequestFactoryProtocol {
+        BlockNetworkRequestFactory {
+            var request = URLRequest(url: self.url)
+
+            let erasRange = historyRange()?.erasRange
+            let params = self.requestParams(accountAddress: self.address, erasRange: erasRange)
+            let info = JSON.dictionaryValue(["query": JSON.stringValue(params)])
+            request.httpBody = try JSONEncoder().encode(info)
+            request.setValue(
+                HttpContentType.json.rawValue,
+                forHTTPHeaderField: HttpHeaderKey.contentType.rawValue
+            )
+            request.httpMethod = HttpMethod.post.rawValue
+            return request
         }
-
-        dependecies.forEach { mergeOperation.addDependency($0) }
-
-        return CompoundOperationWrapper(
-            targetOperation: mergeOperation,
-            dependencies: dependecies
-        )
     }
 
-    private func createSubscanQueryOperation<T>(
-        address: String,
-        moduleName: String,
-        callName: String,
-        mapper: AnyMapper<JSON, T>,
-        reducer: AnyReducer<T, Set<AccountId>>
-    ) -> BaseOperation<Set<AccountId>> {
-        let url = subscanBaseURL
-            .appendingPathComponent(SubscanApi.extrinsics)
-
-        let params = SubscanQueryParams(address: address, moduleName: moduleName, callName: callName)
-
-        return SubscanQueryService(
-            url: url,
-            params: params,
-            mapper: mapper,
-            reducer: reducer,
-            initialResultValue: [],
-            subscanOperationFactory: subscanOperationFactory,
-            operationManager: operationManager
-        ).longrunOperation()
+    private func createResultFactory() -> AnyNetworkResultFactory<[AccountId]> {
+        AnyNetworkResultFactory<[AccountId]> { data in
+            guard
+                let resultData = try? JSONDecoder().decode(JSON.self, from: data),
+                let nodes = resultData.data?.query?.eraValidatorInfos?.nodes?.arrayValue
+            else { return [] }
+
+            let addressFactory = SS58AddressFactory()
+            let validators = nodes
+                .compactMap { SQEraValidatorInfo(from: $0) }
+                .compactMap { validatorInfo -> AccountAddress? in
+                    let contains = validatorInfo.others.contains(where: { $0.who == self.address })
```

Comment:
Why do we need this filter here?

Reply:
Subquery returns array of items as string where could be others.who != queryAddress